regionmask/core/mask.py

```diff
@@ -297,12 +322,23 @@ def _extract_lon_lat(lon_or_obj, lat, lon_name, lat_name):
     return lon, lat
 
 
-def _create_xarray(mask, lon, lat, lon_name, lat_name):
+def _create_xarray(mask, lon, lat, lon_name, lat_name, is_unstructured):
     """create an xarray DataArray"""
 
     # create the xarray output
-    coords = {lat_name: lat, lon_name: lon}
-    mask = xr.DataArray(mask, coords=coords, dims=(lat_name, lon_name), name="region")
+    if is_unstructured:
+        cell_name = lat.dims[0]
+        mask = xr.DataArray(
+            mask,
+            coords={cell_name: lat.coords[cell_name]},
```

Comment:
Should this not be
```suggestion
            coords={lat_name: lat, lon_name: lon},
```
? So we get
```

<xarray.DataArray 'region' (cells: 3)>
array([1, 2, 3])
Coordinates:
    lat      (cells) int64 1 2 3
    lon      (cells) int64 1 2 3
Dimensions without coordinates: cells
```

and not
```
<xarray.DataArray 'region' (cells: 3)>
array([1, 2, 3])
Coordinates:
  * cells    (cells) int64 0 1 2
```